### PR TITLE
Update AWS docs to use Disable Source/Dest. Check instead of IP-in-IP

### PR DESCRIPTION
**Pull request by matthewdupre**
I think this covers all the relevant docs, but it's possible I've missed some (particularly if there's anything outside this repository).

### CODE REVIEW

File: docs/calico-with-docker/AWS.md [hunk 1]
```diff
@@ -190,33 +190,38 @@ aws ec2 run-instances \
...
   --security-group-ids $SECURITY_GROUP_ID \
   --user-data file://<PATH_TO_CLOUD_CONFIG>/user-data-others
-#  --subnet $SUBNET_ID 
+#  --subnet $SUBNET_ID
 #  Include the subnet param above if using a non-default VPC
 ```
 
+Finally, disable `Source/Dest. Check` to allow containers to talk between hosts.  You can disable this with the CLI, or in the `Networking` part of the instances' right click menus.  Run the following for each server:
+```
+aws ec2 modify-instance-attribute --instance-id <INSTANCE_ID> --source-dest-check "{\"Value\": false}"
```
Comment (caseydavenport) [inline docs/calico-with-docker/AWS.md]: It doesn't look like we explain where to get <INSTANCE_ID> from - we should probably mention where that value comes from.

Also, we should use `$INSTANCE_ID` so that this command can be copied / pasted.

File: docs/FAQ.md [hunk 2]
```diff
@@ -96,10 +96,17 @@ The command will need to be run each time the host is restarted.
...
 
-### Can I run Calico in a public cloud environment? 
+### Can I run Calico in a public cloud environment?
 Yes.  If you are running in a public cloud that doesn't allow either L3 peering or L2 connectivity between Calico hosts then you can specify the `--ipip` flag your Calico IP pool:
 ```
 ./calicoctl pool add <CIDR> --ipip --nat-outgoing
 ```
 Calico will then route traffic between Calico hosts using IP in IP.
+
+In AWS, you can instead disable `Source/Dest. Check` as long as all your instances are in the same subnet of your VPC.  This will provide the best performance.  You can disable this with the CLI, or in the `Networking` part of the instances' right click menus.
```
Comment (TrimBiggs) [inline docs/FAQ.md]: Can you say how this will affect the deployment?

Also (nitpick) the last sentence here is a bit wordy, maybe change the second part to:
"...or you can right click an instance in the EC2 console and choose `Change Source/Dest. Check` under `Networking`."

File: docs/calico-with-docker/AWS.md (hunk 1 above)
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: Agree with Casey.  Make sure to have `$INSTANCE_ID_1` and `$INSTANCE_ID_2` after creating each instance.  Probably easiest to just have both modify-instance commands written here for copying and pasting.

File: docs/calico-with-docker/AWS.md [hunk 3]
```diff
@@ -190,33 +190,38 @@ aws ec2 run-instances \
...
 ```
 
 There are two worked examples you can follow: Calico as a Docker network
-plugin, or Calico without Docker networking.  Select the instructions based on 
+plugin, or Calico without Docker networking.  Select the instructions based on
 the networking option that you chose for the cloud config in step (3).
 
 > In the worked example, be sure to follow the additional instructions for
-configuring `ipip` and `nat-outgoing`. 
+configuring `ipip` and `nat-outgoing`.
```
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: This isn't necessarily true anymore if we don't require `ipip`.

File: docs/calico-with-docker/AWS.md [hunk 4]
```diff
@@ -283,38 +288,39 @@ curl http://<host public ip>:80
...
 ```
 Repeat for all relevant Calico interfaces on the host.
 
-Note that the value used is 8981.  This is because Calico traffic will flow 
-through the IP-in-IP tunnel.  When packets enter the IP-in-IP tunnel, an IP 
+If you're using IP-in-IP, instead use a value of 8981 throughout.
+This is because Calico traffic will flow
+through the IP-in-IP tunnel.  When packets enter the IP-in-IP tunnel, an IP
 header of length 20 is added to the packet, summing to a total size of 9001.
+You'll also need to set the MTU of the `tunl0` interface to 8981.
```
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: This is probably going away in the next couple weeks, but for simplicity just add this to the line above:

"Repeat for all relevant Calico interfaces and the `tunl0` interface on the host."

File: docs/calico-with-docker/AWS.md [hunk 5]
```diff
@@ -323,20 +329,20 @@ docker run -v /usr/local/bin:/target jpetazzo/nsenter
...
 ```
 
-Finally, modify the MTU of the Calico interface on the container. If you 
-specified an interface when adding the container to Calico, change `eth1` to be 
+Finally, modify the MTU of the Calico interface on the container. If you
+specified an interface when adding the container to Calico, change `eth1` to be
 the name of the interface you passed in.
 ```
-sudo ifconfig eth1 mtu 8981
+sudo ifconfig eth1 mtu 9001
```
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: Why is this changed?

File: docs/calico-with-docker/without-docker-networking/README.md [hunk 6]
```diff
@@ -63,13 +63,25 @@ You should see output like this on each node
...
-If you are running in the cloud, you will need to first configure an IP Pool
+If you are running in a cloud other than AWS, you will need to first configure an IP Pool
 with the `ipip` and `nat-outgoing` options.
 
 On either node:
 
     calicoctl pool add 192.168.0.0/16 --ipip --nat-outgoing
-    
+
+Instead, if you are running in AWS and all your endpoints are in the same subnet of your VPC,
```
Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: I don't think "Instead" is necessary here.

File: docs/calico-with-docker/without-docker-networking/README.md [hunk 7]
```diff
@@ -63,13 +63,25 @@ You should see output like this on each node
...
+disable Source/Dest. checks on each instance.  You can disable this with the CLI,
+or in the `Networking` part of the instances' right click menus.
```
Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: Same comment as previous reference to this.

Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: Might just be cleaner to include this line and the one below by saying:

"You can disable this by right-clicking on the instance from the EC2 console and choosing `Change Source/Dest. Check`, or by running the following AWS CLI command for each instance:"

File: docs/cni/kubernetes/GCE.md [hunk 8]
```diff
@@ -73,7 +73,7 @@ gcloud compute instances create \
   --image-project coreos-cloud \
   --image coreos-stable-835-9-0-v20151208 \
   --machine-type n1-standard-1 \
-  --metadata-from-file user-data=cloud-config/master-config.yaml
+  --metadata-from-file user-data=cloud-config/master-config-ipip.yaml
```
Comment (TrimBiggs) [inline docs/cni/kubernetes/GCE.md]: Did you remove or modify the master-config.yaml?

Comment (TrimBiggs) [inline docs/cni/kubernetes/GCE.md]: Ah, got it, nvm.  Removing comment.

Comment (TrimBiggs) [inline docs/cni/kubernetes/GCE.md]: Is there somewhere that we want to recommend using the `master-config.yaml` instead of `master-config-ipip.yaml` where `ipip` is not in use?

File: docs/calico-with-docker/without-docker-networking/README.md [hunk 9]
```diff
@@ -63,13 +63,25 @@ You should see output like this on each node
 
 If you are not running in the cloud you may skip this step and jump to step 4.
 
-If you are running in the cloud, you will need to first configure an IP Pool
+If you are running in a cloud other than AWS, you will need to first configure an IP Pool
```
Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: I think it would look nicer if you had two subsections now that the cloud services would use different commands.

Maybe change the main section to just be **Running in the Cloud**, then have subsections of **GCE / Digital Ocean**  and **AWS**

File: docs/cni/kubernetes/GCE.md (hunk 8 above)
Comment (matthewdupre) [inline docs/cni/kubernetes/GCE.md]: Yes: all the other references to this file; they just don't show up in the diff because they didn't need changing.

Comment (TrimBiggs) [inline docs/cni/kubernetes/GCE.md]: Gotcha.

File: docs/calico-with-docker/AWS.md (hunk 5 above)
Comment (matthewdupre) [inline docs/calico-with-docker/AWS.md]: No need to leave room for IPIP by default (and a potential 0.2% performance gain!!!).  Looking at it again though, the wording above is poor and doesn't allow for people turning IPIP on (which TBH I'd rather rip out entirely, but anyway).

I'll just take the 0.2% hit I think...

Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: Ah, I see, I think I misread the above. I think this is fine as 9001, but good to note that here should be 8981 also if using ip-ip.

File: docs/FAQ.md (hunk 2 above)
Comment (matthewdupre) [inline docs/FAQ.md]: Nitpick: agreed, active is better.

As for how it will affect, I'm going to assert that I have: it "will provide the best performance".  If we're going to explain anything in more detail, it should be IPIP.  I don't want to create FUD where there's really no grounds for it by needlessly explaining things.

**Comment by matthewdupre** [comment]
@TrimBiggs back to you

File: docs/calico-with-docker/AWS.md [hunk 10]
```diff
@@ -283,38 +293,39 @@ curl http://<host public ip>:80
...
 ```
 Repeat for all relevant Calico interfaces on the host.
 
-Note that the value used is 8981.  This is because Calico traffic will flow 
-through the IP-in-IP tunnel.  When packets enter the IP-in-IP tunnel, an IP 
+If you're using IP-in-IP, instead use a value of 8981 throughout.
+This is because Calico traffic will flow
+through the IP-in-IP tunnel.  When packets enter the IP-in-IP tunnel, an IP
 header of length 20 is added to the packet, summing to a total size of 9001.
+Repeat for all relevant Calico interfaces and the tunl0 interface on the host.
```
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: Ah, when I made my comment, I meant to modify the sentence 6 lines up, just below the command.  I think you can just use this sentence there and get rid of it here.

File: docs/calico-with-docker/AWS.md [hunk 11]
```diff
@@ -173,13 +173,15 @@ aws ec2 run-instances \
   --key-name mykey \
   --security-group-ids $SECURITY_GROUP_ID \
   --user-data file://<PATH_TO_CLOUD_CONFIG>/user-data-first
-#  --subnet $SUBNET_ID 
+#  --subnet $SUBNET_ID
 #  Include the subnet param above if using a non-default VPC
+
+INSTANCE_ID_1=<InstanceId>
```
Comment (TrimBiggs) [inline docs/calico-with-docker/AWS.md]: There should be a note either be a comment added above here to tell the used to copy the instance ID into this command, or a line above the command to also save the returned instance-id from the output.

This goes for `INSTANCE_ID_2` below also.

**Comment by TrimBiggs** [comment]
Two more comments, then should be good to go

File: docs/calico-with-docker/AWS.md (hunk 11 above)
Comment (matthewdupre) [inline docs/calico-with-docker/AWS.md]: TBH I'm not a fan of this syntax, I think having these additional steps just makes the doc longer (when the variables are generally used once), and people still have to copy the values around.  Every extra sentence in a doc reduces the chance that people will actually follow all the other ones by some tiny percentage, unmeasurable individually but very clear in aggregate.

If we want our docs to be shell scripts, then they should be shell scripts...

Comment (caseydavenport) [inline docs/calico-with-docker/AWS.md]: We've generally tried to keep the large commands as copy-and-pasteable as possible - this makes it less frustrating for people who are trying to follow the guide.

e.g:
- copy the instance ID and export it, copy this command and execute it.

rather than:
- Copy this command somewhere.  Copy the instance ID somewhere. Delete the bit that was a template, fill it in with the instance ID.  Copy and paste that into the terminal (or similar steps).

Especially when you need to capture a bit of configuration in one part of the guide, and then use it again later on, environment variables help with that.  We've received feedback from various docs testers, including @lwr20 that the environment variable method is a better UX.

**Comment by matthewdupre** [comment]
@TrimBiggs OK, tweaked, rebased and squashed.

**Comment by TrimBiggs** [comment]
LGTM

**Comment by matthewdupre** [comment]
Found a missed doc.  It's not the best wording, but it's compact, and these docs are starting to get far too long (considering how many of them cover the same material!)

File: docs/calico-with-docker/docker-network-plugin/README.md [hunk 12]
```diff
@@ -128,6 +128,12 @@ specific environment, you may need to choose different CIDRs.
 So, once you have decided which type of network to create, following the
 appropriate instructions for one of *a)*, *b)*, *c)* or *d)*.
 
+For AWS, chose a) or c), and Disable Source/Dest. checks on your instances with
+the following EC2 CLI command or by right clicking the Instance, and selecting
```
Comment (TrimBiggs) [inline docs/calico-with-docker/docker-network-plugin/README.md]: Nitpick: Instance -> instance

File: docs/calico-with-docker/docker-network-plugin/README.md [hunk 13]
```diff
@@ -128,6 +128,12 @@ specific environment, you may need to choose different CIDRs.
...
+it from the Networking submenu.
+
+    aws ec2 modify-instance-attribute --instance-id $INSTANCE_ID --source-dest-check "{\"Value\": false}"
```
Comment (TrimBiggs) [inline docs/calico-with-docker/docker-network-plugin/README.md]: `$INSTANCE_ID` -> `<instance_id>`

File: docs/calico-with-docker/docker-network-plugin/README.md [hunk 14]
```diff
@@ -128,6 +128,12 @@ specific environment, you may need to choose different CIDRs.
 So, once you have decided which type of network to create, following the
 appropriate instructions for one of *a)*, *b)*, *c)* or *d)*.
 
+For AWS, chose a) or c), and Disable Source/Dest. checks on your instances with
+the following EC2 CLI command or by right clicking the Instance, and selecting
+it from the Networking submenu.
```
Comment (TrimBiggs) [inline docs/calico-with-docker/docker-network-plugin/README.md]: Maybe use the specific name of the option to click instead of "it"? It's not called Disable Source/Dest. here, I think it's `Change Source/Dest. Check`.

File: docs/calico-with-docker/docker-network-plugin/README.md (hunk 12 above)
Comment (TrimBiggs) [inline docs/calico-with-docker/docker-network-plugin/README.md]: "... right clicking the instance **in the EC2 console**, ..." ?

File: docs/calico-with-docker/without-docker-networking/README.md [hunk 15]
```diff
@@ -63,13 +63,26 @@ You should see output like this on each node
 
 If you are not running in the cloud you may skip this step and jump to step 4.
 
-If you are running in the cloud, you will need to first configure an IP Pool
-with the `ipip` and `nat-outgoing` options.
+### 3.1 AWS
+
+If all your endpoints are in the same subnet of your VPC, disable Source/Dest. checks on each instance.  You can disable this with the CLI, or right click the instance in the EC2 console and `Change Source/Dest. Check` from the `Networking` submenu.
+
+On a node with the AWS CLI installed, for each instance:
```
Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: Sorry didn't mention this before, but this could be on any machine with AWS CLI installed, doesn't have to be one of the nodes.  Could just say "Using the AWS CLI..."

File: docs/calico-with-docker/without-docker-networking/README.md [hunk 16]
```diff
@@ -63,13 +63,26 @@ You should see output like this on each node
...
+
+    aws ec2 modify-instance-attribute --instance-id <INSTANCE_ID> --source-dest-check "{\"Value\": false}"
+
+And then configure the `nat-outgoing` option on either Calico node:
```
Comment (TrimBiggs) [inline docs/calico-with-docker/without-docker-networking/README.md]: Maybe this could be clearer about running regardless of setting the Source/Dest. Check in CLI or in the EC2 console, I'd be worried people might think it is just when running the CLI.

Something like: "After disabling Source/Dest. Checks, configure the `nat-outgoing` option on your Calico IP pool from any Calico node:"

File: docs/cni/kubernetes/AWS.md [hunk 17]
```diff
@@ -212,10 +212,15 @@ aws ec2 run-instances \
...
   --security-group-ids $SECURITY_GROUP_ID \
   --user-data file://cloud-config/node-config.yaml
-#  --subnet $SUBNET_ID 
+#  --subnet $SUBNET_ID
 #  Include the subnet param above if using a non-default VPC
 ```
 
+Finally, disable `Source/Dest. Check` on each instance (including the master) to allow routing between pods without needing IP in IP.  All instances must be in the same subnet.  You can do this with the CLI, or in the `Networking` part of the instances' right click menus.
+```
+aws ec2 modify-instance-attribute --instance-id $INSTANCE_ID --source-dest-check "{\"Value\": false}"
```
Comment (TrimBiggs) [inline docs/cni/kubernetes/AWS.md]: The `$INSTANCE_ID` environment variables are not defined in this doc.  You should add these like you did in the  `docs/AWS.md` file and probably modify this command in a similar way to have each necessary `aws cli` call.

**Comment by TrimBiggs** [comment]
Made another pass and added some more comments, should be good once those are in.

**Comment by matthewdupre** [comment]
If you ask me, the value gained by endlessly tweaking the formatting of environment variables is less than the value gained by publishing a day earlier.  Particularly considering most of the rest of the doc doesn't conform anyway.  I've made the updates, please feel free to make any further adjustments you'd like yourself.

**Comment by TrimBiggs** [comment]
I think a few of my comments were nitpicky (which I noted) but the majority were important to get in for clarity and uniformity in the docs.  I'm not quite sure what point you're trying to make with your comment, but the better our docs are, the better the user experience is for our customers, and my opinion is that we should make our docs as easy to understand as possible with little hassle.

That being said, this LGTM apart from merge conflicts.
